examples/ksIterate.c

```diff
@@ -0,0 +1,43 @@
+/**
+ * @file
+ *
+ * @brief
+ *
+ * @copyright BSD License (see LICENSE.md or https://www.libelektra.org)
+ */
+
+#include <kdb.h>
+#include <stddef.h>
+#include <stdio.h>
+
+//! [iterate for]
+void iterate_for (KeySet * ks)
+{
+	Key * cur;
+	for (cursor_t cursor = 0; (cur = ksAtCursor (ks, cursor)) != NULL; ++cursor)
+	{
+		printf ("%s\n", keyName (cur));
+	}
+}
+//! [iterate for]
+
+//! [iterate while]
+void iterate_while (KeySet * ks)
+{
+	cursor_t cursor = 0;
+	Key * cursor_key;
+
+	while ((cursor_key = ksAtCursor (ks, cursor)) != 0)
+	{
+		printf ("%s\n", keyName (cursor_key));
+		cursor++;
```

Comment:
As examples might be copied without looking at it, better to have `++cursor`. Every modern compiler will produce the same code here but you never know.